src/App.js

```diff
@@ -1,19 +1,49 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [messages, setMessages] = useState(chatMessages);
+  const [likeCount, setLikeCount] = useState(0);
+
+  const handleLikeUpdate = (id) => {
+    setMessages(messages => {
+      return messages.map(message => {
+        if (id === message.id) {
+          return {
+            ...message,
+            liked: !message.liked,
+          }
```

Comment:
👍 Flipping the `liked` value here keeps that logic closer to where the state is actually maintained, and with code that needs to "know" what it means to toggle the liked status of a message. Note that we could even move this logic further out of the component so that React doesn't need to know the business rules of these changes. We mostly want to keep React focused on the UI, while flipping the message liked value isn't really a UI concern. React could use a helper to perform this operation without knowing the details of its implementation.